### ISSUE TITLE
Error 400 - HTTP/1.1 400 Bad Request

### ISSUE THREAD

**Issue by etriantafyllidis** [GraesonB/ChatGPT-Wrapper-For-Unity#13]
Hi @GraesonB,

So after further experimenting with the wrapper, I noticed a strange error, namely:
`Error 400 - HTTP/1.1 400 Bad Request`

This error occurs after a good amount of back-and-forth prompts between the user and ChatGPT (3.5-Turbo). 
Essentially, this occurs after I converse with ChatGPT for a while (~30+ prompts back-and-forth).

I thought this could be due to either of the following reasons:
1. Now, from what I understand, the `max_token` size indicates the maximum number of tokens per response, not for the entire conversation, unless I am mistaken. If that is the case, then I should not be "out of tokens" over the duration of the conversation. 
2. Then I thought this could be an internet connection issue, although that was not the case as I got different errors when switching off the internet adapter to quickly test for it. 
3. Nor do I think this is due to sending messages too quickly as the error `Too many requests` pops up.
4. Could this be due to using an API key that is part of a free i.e. non-pay plan? 

In short, I dismissed 2 and 3, I am not sure about 1 or 4.
I find this quite interesting and I am still investigating it. Just wanted to reach out perhaps if you have stumbled on this error :)

My apologies if I am missing something, thank you again for a wonderful wrapper!

**Comment by GraesonB** [comment]
Hi again!

My guess is that this is likely a token issue. To my knowledge there aren't any restrictions on API keys, as long as there are funds on your account (I believe everyone starts with $18) you should have no problem sending API requests.

So the way ChatGPT's API works is it sends the entire conversation with every new message because it needs the full conversation for context. So as a conversation increases in length, the closer you get to ChatGPT's token limit (being 4096). I currently have a solution for this that can increase the length of your conversation well past your 30 prompts, but it's a little more involved and harder to package neatly for other people to use. If you're interested in me giving you the run down, I'd be happy to. My email is graesonbergen@gmail.com.

Cheers!

**Comment by etriantafyllidis** [comment]
Great! Thank you! I will follow up on your email.

I just found this by the way:
[https://help.openai.com/en/articles/7039783-chatgpt-api-faq](url)

> What's the rate limits for the ChatGPT API?
> Free trial users: 20 RPM 40000 TPM
> Pay-as-you-go users (first 48 hours): 60 RPM 60000 TPM
> Pay-as-you-go users (after 48 hours): 3500 RPM 90000 TPM
> 
> RPM = requests per minute
> TPM = tokens per minute

So perhaps the token size limit may not be the issue, as according to the post above, it is 40k tokens / per minute. In any case, it could very well be a token issue, if I find the underlying reason why this is happening I will update the post. Thanks for all your time :)

**Comment by GraesonB** [comment]
Ah I was mistaken, I think you're right! It's probably the TPM restriction. Anyways the solution I mentioned above will also reduce your TPM, so feel free to email me any time.

**Comment by dev-yellowpanda** [comment]
I got this issue as well and was able to fix it by reducing the "Max Tokens" variable to 256 in the inspector
